src/custom/utils/metadata.ts

```diff
@@ -0,0 +1,38 @@
+interface IMetadata {
+  kind: MetadataKind
+  version: string
+}
+
+export enum MetadataKind {
+  REFERRAL = 'referrer',
+}
+
+export type Metadata = {
+  [MetadataKind.REFERRAL]: IReferralMetadata
+}
+
+export type MetadataDoc = {
+  version: string
+  appCode?: string
+  metadata: Metadata
+}
+
+export interface IReferralMetadata extends IMetadata {
+  referrer: string
+}
+
+export const DEFAULT_APP_CODE = 'CowSwap'
+
+export const generateReferralMetadataDoc = (referralAddress: string): MetadataDoc => {
```

Comment:
I'm thinking about an alternative behaviour.
What about instead of storing the "empty" doc, we use this function to generate it.
So, having all parameters optional, when everything is unset, it returns the doc with empty metadata.

This way we have one less place to modify when the doc version/format is updated.

What do you think?

Reply:
Liked the idea. I added a `generateMetadataDoc` function so we can keep any doc updates in one place.